gst/nnstreamer/tensor_query/tensor_query_client.c

```diff
@@ -727,7 +727,8 @@ gst_tensor_query_client_chain (GstPad * pad,
   data_h = g_async_queue_timeout_pop (self->msg_queue,
       self->timeout * G_TIME_SPAN_MILLISECOND);
   if (data_h) {
-    self->requested_num--;
+    if (self->requested_num > 0)
```

Comment:
need { } in lines 730 to 769?

Reply:
yes, pushing buffer to downstream element occurs when a data processed from query-server in the queue.